src/mixer/audio.ts

```diff
@@ -323,25 +343,46 @@ export class AudioEngine extends ((EngineEmitter as unknown) as {
     this.micMixGain.gain.value = value;
   }
 
-  getLevel(source: LevelsSource) {
+  getLevels(source: LevelsSource, stereo: boolean) {
+    let analysisBuffer = new Float32Array(ANALYSIS_FFT_SIZE);
+    let analysisBuffer2 = new Float32Array(ANALYSIS_FFT_SIZE);
     switch (source) {
       case "mic-precomp":
-        this.micPrecompAnalyser.getFloatTimeDomainData(this.analysisBuffer);
+        this.micPrecompAnalyser.getFloatTimeDomainData(analysisBuffer, analysisBuffer2);
         break;
       case "mic-final":
-        this.micFinalAnalyser.getFloatTimeDomainData(this.analysisBuffer);
+        this.micFinalAnalyser.getFloatTimeDomainData(analysisBuffer, analysisBuffer2);
         break;
       case "master":
-        this.streamingAnalyser.getFloatTimeDomainData(this.analysisBuffer);
+        this.streamingAnalyser.getFloatTimeDomainData(analysisBuffer, analysisBuffer2);
+        break;
+      case "player-0":
+        this.player0Analyser.getFloatTimeDomainData(analysisBuffer, analysisBuffer2);
+        break;
+      case "player-1":
+        this.player1Analyser.getFloatTimeDomainData(analysisBuffer, analysisBuffer2);
+        break;
+      case "player-2":
+        this.player2Analyser.getFloatTimeDomainData(analysisBuffer, analysisBuffer2);
         break;
       default:
         throw new Error("can't getLevel " + source);
     }
-    let peak = 0;
-    for (let i = 0; i < this.analysisBuffer.length; i++) {
-      peak = Math.max(peak, Math.abs(this.analysisBuffer[i]));
+    let peakL = 0;
+    for (let i = 0; i < analysisBuffer.length; i++) {
+      peakL = Math.max(peakL, Math.abs(analysisBuffer[i]));
+    }
+    peakL = 20 * Math.log10(peakL);
+
+    if (stereo) {
+      let peakR = 0;
+      for (let i = 0; i < analysisBuffer2.length; i++) {
+        peakR = Math.max(peakR, Math.abs(analysisBuffer2[i]));
```

Comment:
What are the performance implications of running stereo analysis every frame?

Reply:
Hmm, will try and test. I'll have a go at making it mono-able for performance reasons in the settings.